sockets.js

```diff
@@ -443,48 +471,55 @@ if (cluster.isMaster) {
 		sockets.clear();
 		channels.clear();
 		subchannels.clear();
+
 		app.close();
 		if (appssl) appssl.close();
+
+		// Let the server(s) finish closing.
+		setImmediate(() => process.exit(0));
 	});
 
 	// this is global so it can be hotpatched if necessary
 	let isTrustedProxyIp = Dnsbl.checker(Config.proxyip);
 	let socketCounter = 0;
 	server.on('connection', socket => {
-		if (!socket) {
-			// For reasons that are not entirely clear, SockJS sometimes triggers
-			// this event with a null `socket` argument.
-			return;
-		} else if (!socket.remoteAddress) {
-			// This condition occurs several times per day. It may be a SockJS bug.
+		// For reasons that are not entirely clear, SockJS sometimes triggers
+		// this event with a null `socket` argument.
+		if (!socket) return;
+
+		if (!socket.remoteAddress) {
+			// SockJS will sometimes fail to cache a connection's IP, protocol,
+			// and address sometimes. When this happens is likely dependent on
+			// the transport being used and what browser's running the client.
+			// TODO: look more into why this happens.
```

Comment:
Have you verified that this bug still happens?

Reply:
Yes
https://github.com/sockjs/sockjs-node/blob/master/src/transport.coffee#L116

Reply:
Hmm. In any case, I don't think we need to document this case that much, do we?

Reply:
Probably no